tests/integration/docarray_v2/test_v2.py

```diff
@@ -106,3 +108,30 @@ def bar(self, docs: DocumentArray[MyDoc], **kwargs) -> DocumentArray[MyDoc]:
         )
         assert docs[0].text == 'hello world'
         assert docs.__class__.document_type == MyDoc
+
+
+def test_different_output_input():
```

Comment:
make sure in CI this is tested with V2 only

Reply:
yes I added a special CI run that install v2 for these tests